src/SWP/Bundle/AnalyticsBundle/EventListener/KernelListener.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Template Engine Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\AnalyticsBundle\EventListener;
+
+use SWP\Bundle\AnalyticsBundle\Model\RequestMetric;
+use SWP\Bundle\AnalyticsBundle\Repository\RequestMetricRepository;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\Stopwatch\Stopwatch;
+use Symfony\Component\Stopwatch\StopwatchEvent;
+
+class KernelListener
+{
+    /**
+     * @var Stopwatch
+     */
+    protected $stopwatch;
+
+    /**
+     * @var StopwatchEvent
+     */
+    protected $stopwatchEvent;
+
+    /**
+     * @var RequestMetricRepository
+     */
+    protected $requestMetricRepository;
+
+    public function setRequestMetricRepository(RequestMetricRepository $requestMetricRepository)
+    {
+        $this->requestMetricRepository = $requestMetricRepository;
+    }
+
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if (!$event->isMasterRequest()) {
+            return;
+        }
+
+        $route = $event->getRequest()->get('_route');
+
+        // Todo: check for configured routes
+        // Ignore web debug toolbar
+        if (null !== $route && '_wdt' !== $route) {
+            $this->stopwatch = new Stopwatch();
+            $this->stopwatch->start($event->getRequest()->getUri());
+        }
+    }
+
+
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        if (null !== $this->stopwatch) {
+            return;
+        }
+
+        $requestUri = $event->getRequest()->getUri();
+
+        // find out if this is a profiler request, if not don't analyze it
+        try {
+            $this->stopwatchEvent = $this->stopwatch->stop($requestUri);
+        } catch (\LogicException $e) {
+            // do nothing, we shouldn't be analyzing this controller
+            return false;
+        }
+
+        $duration = $this->stopwatchEvent->getDuration();
+        $route = $event->getRequest()->get('_route');
+
+        // Route is null here in test environment
```

Comment:
Why it's like that? If there can be a request without route then better would be to move that check also to top. There is no point to play with other things when at end you can't save it.